simulariumio/data_objects/agent_data.py

```diff
@@ -178,22 +179,34 @@ def get_type_ids_and_mapping(self) -> Tuple[np.ndarray, Dict[str, Any]]:
         last_tid = 0
         for time_index in range(total_steps):
             for agent_index in range(len(self.types[time_index])):
-                tn = self.types[time_index][agent_index]
-                if len(tn) == 0:
+                type_name = self.types[time_index][agent_index]
+                if len(type_name) == 0:
                     continue
-                if tn not in type_id_mapping:
+                if type_name not in type_id_mapping:
                     tid = last_tid
                     last_tid += 1
-                    type_id_mapping[tn] = tid
-                    type_name_mapping[str(tid)] = {"name": tn}
+                    type_id_mapping[type_name] = tid
+                    type_name_mapping[str(tid)] = {"name": type_name}
+                    has_subpoints = self.n_subpoints[time_index][agent_index] > 0
                     if (
-                        tn in self.display_data
-                        and not self.display_data[tn].is_default()
+                        type_name in self.display_data
```

Comment:
I kinda struggled to understand this part of the code because I kept thinking `display_data` was of type `DisplayData` when it's actually a mapping of agent names to their `DisplayData`. I know this isn't new to this PR but I think it'd be helpful to rename `this.display_data` as `this.display_data_map`.

Reply:
It probably would be a little clearer, but it would make this a breaking change and I'd prefer not to do that